modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/CredentialApiUtil.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.orchestrator.utils;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.modules.credential.AbstractCredentialInfo;
+import org.eclipse.xpanse.modules.credential.CredentialDefinition;
+import org.eclipse.xpanse.modules.credential.CredentialVariable;
+import org.eclipse.xpanse.modules.credential.enums.CredentialType;
+import org.eclipse.xpanse.modules.models.enums.Csp;
+import org.eclipse.xpanse.orchestrator.OrchestratorPlugin;
+import org.eclipse.xpanse.orchestrator.OrchestratorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+/**
+ * CredentialApiUtil class.
+ */
+@Slf4j
+@Component
+public class CredentialApiUtil implements ApplicationRunner {
+
+    private final String openapiPath;
+    private final Integer port;
+
+    @Autowired
+    OrchestratorService orchestratorService;
+    @Autowired
+    OpenApiUtil openApiUtil;
+
+    /**
+     * CredentialApiUtil.
+     */
+    @Autowired
+    public CredentialApiUtil(
+            @Value("${openapi.path:openapi/}") String openapiPath,
+            @Value("${server.port:8080}") Integer port) {
+        this.openapiPath = openapiPath;
+        this.port = port;
+    }
+
+    @Override
+    public void run(ApplicationArguments args) {
+        Map<Csp, OrchestratorPlugin> cspOrchestratorPluginMap = orchestratorService.pluginMap();
+        OrchestratorPlugin huaweiOrchestratorPlugin = cspOrchestratorPluginMap.get(Csp.HUAWEI);
```

Comment:
We should traverse all Csps those implement OrchestratorPlugin instead of using fixed Cps.HUAWEI and generate their own API document for each Csp with name ${cspName}_credentialApi.html

Reply:
ok, thanks, i adjust like this.